### PR TITLE
fix(highlight): limit special emoji handing to where it's relevant (#1097)

### PR DESCRIPTION
**Pull request by guyguy2001**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -256,15 +256,21 @@ export function calculateEditorColFromVimScreenCol(
...
+        if (line[currentCharIdx] === "\t") {
+            currentVimCol += tabSize - (currentVimCol % tabSize);
+            currentCharIdx++;
+        }
+        else if (useBytes) {
+            const bytes = getBytesFromCodePoint(line.codePointAt(currentCharIdx));
+            currentVimCol += bytes;
+            // Characters which take 4 bytes also take 2 string indices.
+            // (Only relevant here since wcwidth returns half of the value to each part)
+            currentCharIdx += bytes === 4 ? 2 : 1;
```
Comment (guyguy2001) [inline src/utils.ts:268]: I understand why this check is relevant only here, and therefore feel comfortable with limiting it here, but I still don't understand this function - why do we want to handle tabs differently? they count as just one column when using `col`.
Also, I don't get _why_ wcwidth is used, but I do understand that it doesn't suffer from the problem emoji had

File: src/utils.ts [hunk 2]
```diff
@@ -256,15 +256,21 @@ export function calculateEditorColFromVimScreenCol(
     let currentCharIdx = 0;
```
Comment (guyguy2001) [inline src/utils.ts:256]: Can't we just use `charidx`, either as an async call or copying from the sources?

**Comment by theol0403** [comment]
Off the top of my head, tabs may be treated as emojis because https://github.com/vscode-neovim/vscode-neovim/blob/48900ba60cb9da2f6da50da782ee9bc9e01a0dca/vim/vscode-options.vim#L34

There was a reason for the author to write it but I currently forget.

**Comment by theol0403** [comment]
Can't help right now, middle of exam season. However, feel free to think critically of the code and if you think you can simplify workarounds/reduce bloated code, that would be appreciated. A lot of these workarounds can either be made more clean, or exist for legacy reasons.

**Comment by guyguy2001** [comment]
These are 2-byte emoji though (`'❥'.length === 1`), and from my understanding they aren't used normally in this calculatio

**Comment by guyguy2001** [comment]
I am trying to look at how to improve the relating code, and might make a PR for that soon.
However, this PR solves the bug, and doesn't _seem_ to cause any new ones, from my minimal testing, so I'll upload tests for it (forgot to do that lol) and will appreciate a merge (unless you want me to sanity it for longer, e.g. using it in a real codebase)

**Comment by guyguy2001** [comment]
I worked with it for just a bit, but it seems stable

**Comment by guyguy2001** [comment]
How can I fix all of the linting messages? `npm run lint --fix` doesn't pass the `fix` to the program

**Comment by theol0403** [comment]
Try `npm run lint && npm run format`

**Review by theol0403 — APPROVED** [review]

**Comment by theol0403** [comment]
Thank you for the PR! I'm not really able to help too much with your questions, because I didn't write this part of the code, but feel free to keep tweaking things and see if you can reach a simpler implementation.

**Pull request merged**